packages/clay-data-provider/src/useResource.ts

```diff
@@ -181,12 +187,27 @@ const useResource = ({
 					).searchParams.toString()
 				);
 				break;
-			case 'string':
-				promise = fetch(
-					getUrlFormat(link, variables),
-					fetchOptions
-				).then(res => res.json());
+			case 'string': {
+				const url = getUrlFormat(link, variables);
+
+				if (CONCURRENT_PROMISES[url]) {
+					promise = CONCURRENT_PROMISES[url];
+					break;
+				}
+
+				promise = CONCURRENT_PROMISES[url] = fetch(url, fetchOptions)
```

Comment:
This seems odd to me to have two `=` within the same expressing. Is this intended?

Reply:
If it is intended, I think its probably best to separate them into two separate expressions since it'd be easy to miss while reading through.

Reply:
I added this because we are unable to support when the `link` API is a function, we are unable to know what the endpoint is. It just adds a reference point for both, do you think it's a problem?

Reply:
I was thinking we should at least separate it into two separate declarations. Would this work?

```jsx
promise = fetch( // ...

CONCURRENT_PROMISES[url] = promise
```

Reply:
Yeah, no problem. I will do this.